pkg/controller/account/account_controller.go

```diff
@@ -253,6 +253,10 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 
 			if !accountHasAwsAccountID(currentAcctInstance) {
 				// before doing anything make sure we are not over the limit if we are just error
+				if totalaccountwatcher.TotalAccountWatcher.Total == 0 {
```

Comment:
I don't really feel like 0 is a whole lot better than 99999.

My suggestion about leaving the `TotalAccountWatcher` global `nil` until it's successfully initialized would require a bigger refactor, but a simpler version meets you halfway to your `AccountsCanBeCreated` field: Add an `Initialized` bool field that you only set to `true` once `UpdateTotalAccounts` has succeeded, and check that here instead of the `Total`.

Reply:
Demonstrated [here](https://github.com/iamkirkbater/aws-account-operator/pull/1)

Reply:
+1 to the suggestion with the `Initialized` bool, it prevents a bug where we're using the Payer Account for the first time and it'll have actually ZERO accounts in it